### PR TITLE
[REVIEW] Accept list of nvstrings for cat() method

### LINKED ISSUE #323: [FEA] support for cat to take multiple nvstrings as input
Currently there is no direct way to actually append a list of nvstrings. Present `nvstrings.cat` takes only one nvstring as input parameter.  So when we would like to do string concatenation in bulk we will have to chain the operation.

Like:

```
>>> import nvstrings
>>> s1 = nvstrings.to_device(['hello', None,'goodbye'])
>>> s2 = nvstrings.to_device(['world','globe', None])
>>> s3 = nvstrings.to_device(['one','two', None])
>>> print(s1.cat(s2,sep=':', na_rep='_').cat(s3,sep='-', na_rep='_')
["hello:world-one","_:globe-two","goodbye:_-_"]
```

But instead if multiple nvstrings could be allowed to be passed to `cat`, it would look something like this:

```
>>> import nvstrings
>>> s1 = nvstrings.to_device(['hello', None,'goodbye'])
>>> s2 = nvstrings.to_device(['world','globe', None])
>>> s3 = nvstrings.to_device(['one','two', None])
>>> print(s1.cat(others=[s2,s3],sep=[':','-'], na_rep=['_']))
["hello:world-one","_:globe-two","goodbye:_-_"]
```

cc: @kkraus14

### PR DESCRIPTION
**Pull request by davidwendt**
Closes #323

### CODE REVIEW

**Comment by davidwendt** [comment]
@beckernick I changed the nvstrings `cat()` parameter to accept nvstrings only in this PR. It previously accepted a list of host strings `['a','b','c']`  I replaced this with accepting either a single nvstrings (existing behavior) or a list of nvstrings `[dstrs1,dstrs2,...]`. If you agree, can you approve this PR?

**Comment by beckernick** [comment]
That change is fine with me, as long as cuDF users are still able to pass a list of host strings that is automatically converted to an nvstrings instance as needed.

It looks to me like inside the `cat` function on StringColumns this is happening in every if/else statement, so we should be fine.

https://github.com/rapidsai/cudf/blob/7639414a3bffd7529c800ec53d8aeb44518c8371/python/cudf/dataframe/string.py#L137

I'll test this expicitly before approving on the Python side, though

**Comment by beckernick** [comment]
@davidwendt this will currently segfault if we pass a host array. We should protect end users with an informative error instead.

File: python/nvstrings.py [hunk 1]
```diff
@@ -878,11 +878,10 @@ def cat(self, others=None, sep=None, na_rep=None):
 
         Parameters
         ----------
-        others : List of str
+        others : nvstrings or list of nvstrings
             Strings to be appended.
```
Comment (beckernick) [inline python/nvstrings.py]: Let's raise an informative error if `others` is a Python list of CPU strings to avoid segfaulting.

Comment (beckernick) [reply python/nvstrings.py]: This is passing cuDF string pytests for `str.cat`, which includes handling a list of Python strings. Pending this safeguard, this LGTM David

Comment (davidwendt) [reply python/nvstrings.py]: Argh, yes. Good catch.

**Comment by beckernick** [comment]
Clear error raised to protect the user. LGTM

**Pull request merged**